### PR TITLE
bump lib dependencies

### PR DESCRIPTION
**Pull request by ktalanda**
- retrofit 2.1.0,
- okHttp 3.5.0
- rxJava 1.2.3
- rxAndroid 1.2.1

### CODE REVIEW

File: library/src/main/java/pl/rspective/voucherify/android/client/api/VoucherifyApi.java [hunk 1]
```diff
@@ -22,7 +23,7 @@
      * @return
      */
     @GET("/client/v1/validate")
-    VoucherResponse validateVoucher(@QueryMap Map<String, String> queryParams);
+    Observable<VoucherResponse> validateVoucher(@QueryMap Map<String, String> queryParams);
```
Comment (polok) [inline library/src/main/java/pl/rspective/voucherify/android/client/api/VoucherifyApi.java]: Why we force users to use Observables ? What if someone doesn't want to use RxJava/Android world, what if someone what to use a normal asynchronous call? I would rather think about 'Call' - https://square.github.io/retrofit/2.x/retrofit/retrofit2/Call.html

The current version of the SDK allows the users to call API in the synchronous, asynchronous or rx way.

**Review by polok — CHANGES_REQUESTED** [review]
The API has to be flexible. We can't force to use Observables.

File: library/src/main/java/pl/rspective/voucherify/android/client/api/VoucherifyApi.java (hunk 1 above)
Comment (ktalanda) [reply library/src/main/java/pl/rspective/voucherify/android/client/api/VoucherifyApi.java]: right, will get back to synchronous retrofit calls, thx :+1:

**Review by polok — CHANGES_REQUESTED** [review]
Now, I like it :).
However, before we merge this. There are two small issues related to our README file. Please, update `proguard` section and `RetrofitError`in the examples as there is no such class in retrofit 2.

**Comment by ktalanda** [comment]
👍

**Review by polok — APPROVED** [review]
👍

**Pull request merged**